### DISCUSSION TITLE
Github Enterprise List all Pull Requests Files operation is giving error 422 on specific page.

### DISCUSSION THREAD

**Discussion by ashutosh150596** [community/community#58757] [Enterprise]
### Select Topic Area

Question

### Body

Hello Everyone,

I'm trying to fetch the Pull Requests Files using the Github Enterprise REST API:

<Github Enterprise URL>/api/v3/repos/{repo/owner}/pulls/{pullnumber}/files

and applying the pagination as:

<Github Enterprise URL>/api/v3/repos/{repo/owner}/pulls/{pullnumber}/files?per_page=100&page=1

Similarly to fetch next page data, I'm increasing the page to 2, then 3 and so on.
Upon doing the above steps I found that the REST API is failed to give data for some specific page, for ex: for the below REST API I fired,

<Github Enterprise URL>/api/v3/repos/myrepo/testrepo/pulls/1/files?per_page=100&page=2

 I got the error:
 <422> Unprocessable Entity
{

    "message": "Server Error: Sorry, this diff is taking too long to generate.",

    "errors": [

        {

            "resource": "PullRequest",

            "field": "diff",

            "code": "not_available"

        }

    ],

    "documentation_url": "[https://docs.github.com/enterprise-server@3.7/rest/reference/pulls#list-pull-requests-files"](https://docs.github.com/enterprise-server@3.7/rest/reference/pulls#list-pull-requests-files%22)

}

I have also tried to read the documentation related to this REST API and provided the details as mentioned in the document but still gets the same error. However, I observed 1 thing that after this page the REST API starts to give the data as usual. So, in our example as I mentioned above, If I fire the next page REST API i.e. 

<Github Enterprise URL>/api/v3/repos/myrepo/testrepo/pulls/1/files?per_page=100&page=3

I'm getting the data successfully but not on the page=2.

Requesting everyone to help if they have encountered the same issues before on Github Enterprise.

Thanks,
Github User.

**Comment by queenofcorgis** [comment]
Welcome to the GitHub Community, @ashutosh150596 , we're happy you're here! You are more likely to get a useful response if you are posting your question in the applicable category. I've gone ahead and moved it for you. Good luck!
